Sources/NIOSSL/SSLCallbacks.swift

```diff
@@ -130,6 +130,22 @@ extension KeyLogCallbackManager {
     }
 }
 
+/// PSK Identity Callback Manager
+public typealias NIOPSKClientIdentityCallback = (NIOSSLSecureBytes?) -> (key: NIOSSLSecureBytes, identity: NIOSSLSecureBytes)
+public typealias NIOPSKServerIdentityCallback = (NIOSSLSecureBytes?, NIOSSLSecureBytes?) -> (key: NIOSSLSecureBytes, identity: NIOSSLSecureBytes)
+
+
+/// An object that provides helpers for clients / servers wanting to expose PSKs to their context.
+public struct PSKIdentityCallbackManager {
```

Comment:
The reason I have left this as public and it's members as public is because when I want to consume these types from, let's say `NIOTLSServer` I get the following errors:
```
Cannot find 'PSKIdentityCallbackManager' in scope
```
or even:
```
'Some type'is inaccessible due to 'internal' protection level
```
With the following code just placed in `NIOTLSServer`:
```swift
func pskCallback(hint: NIOSSLSecureBytes?, identity: NIOSSLSecureBytes?) -> PSKIdentityResponse {
    var psk = NIOSSLSecureBytes()
    var pskIdentity = NIOSSLSecureBytes()
    psk.append(contentsOf: "hello".utf8)
    pskIdentity.append(contentsOf: "world".utf8)
    return PSKIdentityResponse(key: psk, identity: pskIdentity)
}

let clientPSKCallbackManager = PSKIdentityCallbackManager(clientCallback: nil, serverCallback: pskCallback)
```
Beings how and application like `NIOTLSServer` would be the consumer here of the PSK functionality, then I leave this block of code in to make sure these types work well here too.  
Thoughts?

Reply:
My proposal is that instead of saving a `PSKIdentityCallbackManager` on the `TLSConfiguration` we have two variables, one with each callback.

Reply:
@Lukasa I wanted to ask about this before I proceeded. Did to take:  
```swift
public typealias NIOPSKClientIdentityCallback = (String, String) -> PSKIdentityResponse
public typealias NIOPSKServerIdentityCallback = (String, String) -> PSKIdentityResponse
```
And add them both as options to configure on `TLSConfiguration` struct?  So `TLSConfiguration` would then have:
```swift
public var pskClientCallback: NIOPSKClientIdentityCallback? = nil

public var pskServerCallback: NIOPSKServerIdentityCallback? = nil
```
Let me know your thoughts?

Reply:
Thank you. Resolved in 516a89f.